resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -86,11 +88,14 @@ private[spark] class ExecutorPodsAllocator(
           s" cluster after $podCreationTimeout milliseconds despite the fact that a" +
           " previous allocation attempt tried to create it. The executor may have been" +
           " deleted but the application missed the deletion event.")
-        Utils.tryLogNonFatalError {
-          kubernetesClient
-            .pods()
-            .withLabel(SPARK_EXECUTOR_ID_LABEL, execId.toString)
-            .delete()
+
+        if (shouldDeleteExecutors) {
+          Utils.tryLogNonFatalError {
+            kubernetesClient
+              .pods()
+              .withLabel(SPARK_EXECUTOR_ID_LABEL, execId.toString)
```

Comment:
We should also add labels `SPARK_APP_ID_LABEL` and `SPARK_ROLE_LABEL`.

Reply:
Ok let's make that configurable too.

Reply:
I think it's fine to do this later, i.e., making sure that we use consistent sets of labels when listing and deleting pods.

Reply:
Actually just realized that `execId` is just an integer incremented whenever a new executor pod gets created. We definitely need to add the two labels to make sure we don't accidentally delete executor pods with the same `execId` from a different application.

Reply:
Correct here is a sample output:
```Labels:    spark-app-selector=spark-application-1543242814683
                spark-exec-id=2
                spark-role=executor
```
Weird that didnt see one job interfering with the other before.

Reply:
I just run multiple jobs in parallel:
```
NAME                            READY     STATUS      RESTARTS   AGE
spark-pi-1543281035622-driver   1/1       Running     0          2m
spark-pi-1543281035622-exec-1   1/1       Running     0          1m
spark-pi-1543281098418-driver   0/1       Completed   0          1m
spark-pi-1543281107591-driver   0/1       Completed   0          57s
```
The long running was not terminated...

Reply:
It seems like this change fixes an existing bug. Should this be a separate PR?

Reply:
@liyinan926 ok. In general it is not a big issue but for tracking purposes it would help.